### PR TITLE
PhotoSwipe in detached window, Shared external window; [issue-3] [issue-2]

### PR DESCRIPTION
**Pull request by JaySunSyn**
See https://github.com/Neovici/cosmoz-image-viewer/issues/3

---

Changes:
- `currentImage` got readOnly; To change an image use `currentImageIndex` (safer when having duplicates in the image array)
- `sizing`, `height`, `nav`, `dots`, `fullscreenOverlay`properties added
- `showInvoiceImage` renamed => `enterFullscreen`

---

Upstreamed changes to 
- https://github.com/JaySunSyn/skeleton-carousel/tree/hybrid
- https://github.com/FabricElements/skeleton-carousel
- https://github.com/Link2Twenty/l2t-paper-slider

---
I first tried to replace it directly with swiper js v4. (https://github.com/JaySunSyn/cosmoz-image-viewer/tree/issue-3)
Ran into some weird behavior on different screen sizes. Docs are still for v3. We may should wait a bit and then replace fullscreen/detached/gallery with swiper.js at once.

### CODE REVIEW

File: cosmoz-image-viewer.js [hunk 1]
```diff
@@ -136,15 +136,52 @@
...
+								this.dispatchEvent(event);
+							})();
+						</script>
+					</body>
+				</html>
+			`);
+			w.document.close();
 			w.document.title = this._('Cosmoz Image Viewer');
+			w.addEventListener('ready', (e) => {
+				var imgs = [
```
Comment (nomego) [inline cosmoz-image-viewer.js]: This should probably be moved to demo/test?

File: demo/index.html [hunk 2]
```diff
@@ -6,13 +6,14 @@
 
 		<title>cosmoz-image-viewer demo</title>
 
-		<script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+		<script src="../../webcomponentsjs/webcomponents-lite.js"></script>
```
Comment (nomego) [inline demo/index.html]: Trailing spaces.. should be auto-fixed with vscode?
Missing vscode settings?

File: demo/index.html [hunk 3]
```diff
@@ -25,18 +26,28 @@
 			<h3>Basic <code>cosmoz-image-viewer</code>.</h3>
 			<demo-snippet>
 				<template>
-					<cosmoz-image-viewer></cosmoz-image-viewer>
+					<dom-bind class="scope">
+						<template is="dom-bind" class="scope">
+							<cosmoz-image-viewer images="[[images]]"></cosmoz-image-viewer>
```
Comment (nomego) [inline demo/index.html]: Two needed?

File: demo/index.html (hunk 2 above)
Comment (JaySunSyn) [reply demo/index.html]: Worked only in a javascript block.

Added now `"files.trimTrailingWhitespace": true`

File: demo/index.html (hunk 3 above)
Comment (JaySunSyn) [reply demo/index.html]: To showcase the shared "detached" window.

**Comment by JaySunSyn** [comment]
Sorry, missed that you reviewed earlier!
Working with PhotoSwipe was pretty smooth tbh. If moving to Swiper makes sense depends on future requirements I'd say. Nice thing about Swiper.js is variety of settings and that you can link two Swiper.js instances which can control each other => gallery view with thumbnails.

**Comment by JaySunSyn** [comment]
I upstreamed some more changes to skeleton-carousel which will be merged hopefully later today. Afterwards I'll push some some changes which should make it be ready to go.

**Comment by JaySunSyn** [comment]
@plequang please check.

**Pull request merged**